Crash when there are utf8 emoji in the title of activity

**Describe the bug**
Crash after stop tracking.

**To Reproduce**
1. Enable automatic GPX export after stop tracking.
2. Start tracking.
3. Set name of the activity for example to "Food 🍛 —> Frog 🐸".
4. Tap to stop.
5. Crash.

```
E/[qarth_debug:]  get PatchStore::createDisableExceptionQarthFile method fail.
E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:355)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:784)
     Caused by: java.lang.IllegalArgumentException: Failed to determine if primary:2562b1e5_Cattani 🍛? —_ Žabka 🐸? (🚦?6👎?_3👍?).gpx is child of primary:...: java.io.FileNotFoundException: Missing file for primary:2562b1e5_Cattani 🍛? —_ Žabka 🐸? (🚦?6👎?_3👍?).gpx at /storage/emulated/0/2562b1e5_Cattani 🍛? —_ Žabka 🐸?.gpx
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:165)
        at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:146)
        at android.content.ContentProviderProxy.openAssetFile(ContentProviderNative.java:625)
        at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1331)
        at android.content.ContentResolver.openOutputStream(ContentResolver.java:1073)
        at android.content.ContentResolver.openOutputStream(ContentResolver.java:1049)
        at de.dennisguse.opentracks.util.ExportUtils.exportTrack(ExportUtils.java:56)
        at de.dennisguse.opentracks.io.file.exporter.ExportService.onHandleWork(ExportService.java:62)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:396)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:387)
        at android.os.AsyncTask$2.call(AsyncTask.java:334)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:784) 
```
The problem are maybe emoji in the filename..?

**Technical information**
 - Device: Honor 8S KSA-LX9
 - OS: Android
 - Version v3.24.1

--- a/src/androidTest/java/de/dennisguse/opentracks/util/FileUtilsTest.java
+++ b/src/androidTest/java/de/dennisguse/opentracks/util/FileUtilsTest.java
@@ -15,14 +15,14 @@
  */
 package de.dennisguse.opentracks.util;
 
+import static org.junit.Assert.assertEquals;
+
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 import java.io.File;
 import java.util.Arrays;
-
-import static org.junit.Assert.assertEquals;
 
 /**
  * Tests for {@link FileUtils}.
@@ -98,6 +98,13 @@ public class FileUtilsTest {
     public void testSanitizeFileName_collapse() {
         String name = "hello//there";
         String expected = "hello_there";
+        assertEquals(expected, FileUtils.sanitizeFileName(name));
+    }
+
+    @Test
+    public void testSanitizeFileName_emoji() {
+        String name = "\uD83C\uDF5B-Food";
+        String expected = "_-Food";
         assertEquals(expected, FileUtils.sanitizeFileName(name));
     }
 

--- a/src/main/java/de/dennisguse/opentracks/util/FileUtils.java
+++ b/src/main/java/de/dennisguse/opentracks/util/FileUtils.java
@@ -146,7 +146,7 @@ public class FileUtils {
         for (int i = 0; i < name.length(); i++) {
             int codePoint = name.codePointAt(i);
             char character = name.charAt(i);
-            if (Character.isLetterOrDigit(character) || codePoint > 127 || isSpecialFat32(character) || character == '.') {
+            if (Character.isLetterOrDigit(character) || isSpecialFat32(character) || character == '.') {
                 builder.appendCodePoint(codePoint);
             } else {
                 builder.append("_");

